chromium_src/components/permissions/permission_request.cc

```diff
@@ -5,20 +5,63 @@
 
 #include "components/permissions/permission_request.h"
 
+#include <vector>
+
+#include "base/containers/contains.h"
+#include "third_party/widevine/cdm/buildflags.h"
+
 #define PermissionRequest PermissionRequest_ChromiumImpl
 #define IsDuplicateOf IsDuplicateOf_ChromiumImpl
+
+// `kWidevine` handled by an override in `WidevinePermissionRequest` and the
+// Brave Ethereum permission has its own permission request prompt.
+#define BRAVE_PERMISSION_REQUEST_GET_MESSAGE_TEXT_FRAGMENT \
+  case RequestType::kBraveEthereum:                        \
+    NOTREACHED();                                          \
+    return std::u16string();                               \
+  case RequestType::kWidevine:                             \
+    NOTREACHED();                                          \
+    return std::u16string();
+
 #include "../../../../components/permissions/permission_request.cc"
+#undef BRAVE_PERMISSION_REQUEST_GET_MESSAGE_TEXT_FRAGMENT
 #undef IsDuplicateOf
 #undef PermissionRequest
 
 namespace permissions {
 
-PermissionRequest::PermissionRequest() = default;
+PermissionRequest::PermissionRequest(
+    const GURL& requesting_origin,
+    RequestType request_type,
+    bool has_gesture,
+    PermissionDecidedCallback permission_decided_callback,
+    base::OnceClosure delete_callback)
+    : PermissionRequest_ChromiumImpl(requesting_origin,
+                                     request_type,
+                                     has_gesture,
+                                     std::move(permission_decided_callback),
+                                     std::move(delete_callback)) {}
 
 PermissionRequest::~PermissionRequest() = default;
 
 bool PermissionRequest::SupportsLifetime() const {
-  return false;
+  static const std::vector<RequestType> excluded_types = {
```

Comment:
nice idea! I think it will be better to store it as a const array on stack:
```
const RequestType kExcludedTypes[] = { ... };
if (base::Contains(kExcludedTypes, request_type())) { ... }
```

Reply:
Changed in cf37bf3